src/main/java/org/candlepin/subscriptions/resource/HostsResource.java

```diff
@@ -80,14 +80,19 @@ public HostsResource(HostRepository repository, PageLinkCreator pageLinkCreator)
     public HostReport getHosts(String productId, Integer offset, Integer limit, String sla,
         String usage, Uom uom, HostReportSort sort, SortDirection dir) {
 
-        Sort sortValue = Sort.unsorted();
         Sort.Direction dirValue = Sort.Direction.ASC;
-
         if (dir == SortDirection.DESC) {
             dirValue = Sort.Direction.DESC;
         }
+        Sort.Order implicitOrder = Sort.Order.by("id");
+        Sort sortValue;
+
         if (sort != null) {
-            sortValue = Sort.by(dirValue, SORT_PARAM_MAPPING.get(sort));
+            Sort.Order userDefinedOrder = new Sort.Order(dirValue, SORT_PARAM_MAPPING.get(sort));
+            sortValue = Sort.by(userDefinedOrder, implicitOrder);
+        }
+        else {
+            sortValue = Sort.by(implicitOrder);
```

Comment:
```java
        Sort sortValue = Sort.by(implicitOrder);

        if (sort != null) {
            Sort.Order userDefinedOrder = new Sort.Order(dirValue, SORT_PARAM_MAPPING.get(sort));
            sortValue = Sort.by(userDefinedOrder, implicitOrder);
        }
```

^^^ What do you think?  Makes it a little more succinct.

Reply:
Done